cerberus-dashboard/src/reducers/authenticationReducer.js

```diff
@@ -108,7 +108,11 @@ export default createReducer(initialState, {
         return Object.assign({}, state, {
             selectedDeviceId: payload.selectedDeviceId,
             shouldDisplaySendCodeButton: state.mfaDevices
-                .filter(device => device.id === payload.selectedDeviceId)[0].requires_trigger
+                .filter(device => device.id === payload.selectedDeviceId)[0].requires_trigger &&
+                !state.mfaDevices.filter(device => device.id === payload.selectedDeviceId)[0].is_push,
+            shouldDisplaySendPushButton: state.mfaDevices
```

Comment:
Trying to parse this code here. Is this setting the default option for users as a push notification if it is enabled?

Reply:
Let me take a look at this as it may no longer be needed. I did some refactoring to this workflow later into my changes. I'll update when I confirm.

Reply:
Ok, I just reviewed this. 

Essentially what this does is determine the layout of the page. Originally I was reusing the "requires_trigger" parameter to help determine this but I refactored that for clarity. My next update will clean this up.